src/ClientData/include/ClientData/ProcessData.h

```diff
@@ -69,7 +69,13 @@ class ProcessData final {
   [[nodiscard]] const std::string& build_id() const;
 
   void UpdateModuleInfos(absl::Span<const orbit_grpc_protos::ModuleInfo> module_infos);
-  void AddOrUpdateModuleInfo(const orbit_grpc_protos::ModuleInfo& module_infos);
+
+  // This method removes all modules with addresses intersecting with module_info.
+  // Some background on this. The service currently does not send any unmap event
+  // to the client, which means that if the client receives a module with mapping
+  // intersecting with exiting mapping the old module was likely unloaded.
```

Comment:
But could it happen that we have absolute addresses in the capture that refer to the unloaded module? Wouldn't it be a problem if we have removed the module from the mapping?

What would happen if we kept both? That then we wouldn't know how to treat absolute addresses that fall in the intersection hence making things even worse?

Reply:
Let me start by saying the proper fix is to have a timeline for module load events but this is out of the scope of this PR.

The most likely use-case is that the  same address is remapped in which case this does not make it worth. 
The second use case is having 2 consecutive modules unmapped and 1 (bigger) module remapped in their place, in which case this PR makes things better since it it removes the second module before mapping the bigger one. Without it looking up addresses after former start of the second module will result in incorrect module/function addresses being used.

The third use case if somehow the mapping ended up intersecting 1 module. This could happen for example if someone unloads a module then calls mmap and then loads different module. In which case with old logic we will end up with the checking the second module in most cases since anything bigger new module address will look only at new module start address. even if it end up not being in the requested address range.

The only use case which makes things a bit worse is when we happen to keep old module around and address lands intersection between start of old module and start of new one. In which case old logic accidentally gets it right.

Reply:
Thanks. I think it would be useful to have a summary of this explanation in a comment, or at the very least in the commit message.

Reply:
done (moved parts of this into commit message)